src/data-table.component.ts

```diff
@@ -647,6 +662,19 @@ export class DataTableComponent implements OnInit, OnDestroy, OnChanges {
     }
   }
 
+  isHeaderCheckboxDisabled() {
+    return Array.from(this.rowSelectionDisabledStates.values()).every((disabledState) => disabledState === true);
+  }
+
+  tableContentChanged() {
```

Comment:
if we had the `@Input` this could be used to check if is provided and invoke it.